controllers/numaresourcesoperator_controller.go

```diff
@@ -581,6 +583,43 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 			return nil, fmt.Errorf("failed to apply (%s) %s/%s: %w", objState.Desired.GetObjectKind().GroupVersionKind(), objState.Desired.GetNamespace(), objState.Desired.GetName(), err)
 		}
 	}
+
+	for _, obj := range r.RTEMetricsManifests.ToObjects() {
+		// Check if the object already exists
+		existingObj := obj.DeepCopyObject().(client.Object)
+		err := r.Client.Get(ctx, client.ObjectKeyFromObject(obj), existingObj)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to get %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
+		}
+		if apierrors.IsNotFound(err) {
+			err := controllerutil.SetControllerReference(instance, obj, r.Scheme)
+			if err != nil {
+				return nil, fmt.Errorf("failed to set controller reference to %s %s: %w", obj.GetNamespace(), obj.GetName(), err)
+			}
+			err = r.Client.Create(ctx, obj)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
+			}
+		} else {
```

Comment:
The only change I would suggest here to make the logic more clear is to encapsulate the `else` in:
`UpdateIfNeeded()`  function
The function won't call `Update()` if the objects are actually equals

```
err := r.Client.Get(ctx, client.ObjectKeyFromObject(obj), existingObj)
		if err != nil && !apierrors.IsNotFound(err) {
			return nil, fmt.Errorf("failed to get %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
		}
		if apierrors.IsNotFound(err) {
			err := controllerutil.SetControllerReference(instance, obj, r.Scheme)
			if err != nil {
				return nil, fmt.Errorf("failed to set controller reference to %s %s: %w", obj.GetNamespace(), obj.GetName(), err)
			}
			err = r.Client.Create(ctx, obj)
			if err != nil {
				return nil, fmt.Errorf("failed to create %s/%s: %w", obj.GetNamespace(), obj.GetName(), err)
			}
		} else {
                           if err := UpdateIfNeeded(ctx, obj, existingObj); err != nil {
                             ....
                           }
}

func UpdateIfNeeded()...
```